iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractIdentifiable.java

```diff
@@ -146,7 +146,7 @@ public String setProperty(String key, String value) {
         } else {
             notifyElementReplaced(key, oldValue, val);
         }
-        return oldValue != null ? (String) oldValue.getValue() : null;
+        return oldValue != null && val.getKey().equals(oldValue.getKey()) ? (String) oldValue.getValue() : null;
```

Comment:
Rename `val` into `newValue` or more explicit
You may add a new method : isSameType ? (as isValueFound)